provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/ProvisioningDeviceClientConfig.java

```diff
@@ -150,4 +157,25 @@ public boolean getUseWebSockets()
     {
         return this.useWebSockets;
     }
+
+    /**
+     * Setter data for custom payload
+     * @param json payload data
+     */
+    public void setPayload(String jsonPayload)
+    {
+        Gson gson = new GsonBuilder().disableHtmlEscaping().create();
+        gson.toJson(this);
+
+        this.jsonPayload = jsonPayload;
+    }
```

Comment:
This is bringing Gson dependency to higher up's. Is it possible to restrict this dependency in deps?